myldr/boot.c

```diff
@@ -77,6 +85,58 @@ int extract_embedded_file(embedded_file_t *emb_file, const char* ext_name, const
     return EXTRACT_OK;
 }
 
+/* Simple magic number reader for MacOS */
+#ifdef __MACH__
+uint32_t read_magic(FILE *obj_file, int offset) {
+  uint32_t magic;
+  fseek(obj_file, offset, SEEK_SET);
+  fread(&magic, sizeof(uint32_t), 1, obj_file);
+  return magic;
+}
+
+/* Duplicate argv with some changes */
```

Comment:
There's no need to create a deep copy of argv, just copy the argv array (including the trailing NULL) and replace argv[0] afterwards. In fact, since we either execve away or die, you might just modify the original argv.

Reply:
Yes, that is currently a bit of a hack and it also doesn't work as it is currently, looking at it.

Reply:
Resolved this, will update pull request with other bits today. Just changing `argv[0]` now.